File: lambda/problema3.hs
{-
Crie uma express ̃ao Lambda que compute o n- ́esimo n ́umero de Fibonacci.
Leia n do teclado

exemplo sintaxe
fatorial :: Int -> Int
fatorial = fix (\f n -> if n == 0 then 1 else n * (f (n-1)))

main = do
  print (fatorial 5)

-}

import Data.Function

fib :: Int -> Int
fib = fix (\f n -> if n==0 then 0 else if n==1 then 1 else (f (n-1)) + (f (n-2)) )

main = do
  print ( (fib 5) )

File: python/problema3.py
'''
3. Crie uma expressão Lambda que compute o n-ésimo número de Fibonacci.
 Leia n do teclado.
'''
num = int(input("entre com numero\n"))
print("Fibbonacci desse numero eh "+str(
(lambda f: f(f)) (lambda f, i=0, fibz=0, fib=1: fibz if i==num else f(f,i+1,fib,fibz+fib))
))
